### ISSUE TITLE
[question] [DataGrid] Is `valueOptions` required to be set on `singleSelect` column types?

### ISSUE THREAD

**Issue by sydneyjodon-wk** [mui/mui-x#14601]
### Order ID or Support key 💳

82849

### Search keywords

valueOptions required singleSelect

### Latest version

- [X] I have tested the latest version

### Affected products

Data Grid

### The problem in depth

Hello 👋 

My team has a `singleSelect` column type that we overrode the edit cell for to make it multi-select and therefore didn't set `valueOptions` because we no longer needed it. This has seemed to work fine, but we just ran into an error being thrown when trying to export to excel:

```
Cannot read properties of undefined (reading 'address')
```
Click the "download as excel" option to repro in this Codesandbox: https://codesandbox.io/p/sandbox/datagrid-singleselect-without-valueoptions-hydgjz?file=%2Fsrc%2FDemo.tsx 

If I add a `valueOptions`, it works again!

I understand that my usage could probably qualify more as a `type: 'string'` column, but `valueOptions` [is marked as optional](https://github.com/mui/mui-x/blob/2bd21e8fd6f3b2ac6d03410975bac6016f7336c0/packages/x-data-grid/src/models/colDef/gridColDef.ts#L328-L330) and it seemed to work without setting it until I ran into this error. So **my question is: Is this particular case an issue with the download to excel or is `valueOptions` actually required on `singleSelect` columns?** If it is the later, would it be possible to change the type definition to reflect that it is required to avoid this type of thing where things are assumed to work without it, but then some case comes up where it throws. Or maybe there's some other issue that I'm missing that's causing this?

Thank you!

### Your environment

See codesandbox: https://codesandbox.io/p/sandbox/datagrid-singleselect-without-valueoptions-hydgjz?file=%2Fsrc%2FDemo.tsx

**Comment by michelengelen** [comment]
Hey @sydneyjodon-wk ... is there a specific reason why you need to keep the column type as `'singleSelect'` ... it works perfectly fine when you set it to `'string'`

**Comment by michelengelen** [comment]
But yes, I do agree that the `valueOptions` property should not be optional.
I'll mark this as a bug and add it to the board for the team to discuss.

**Comment by sydneyjodon-wk** [comment]
Yep! I know it works fine as string, I'm just worried about our consumers doing something similar and using `'singleSelect'` without `valueOptions` assuming that it works and then having it throw unexpectedly. Thanks for updating that!
